src/vos/vos_dtx.c

```diff
@@ -2192,6 +2192,4 @@ vos_dtx_cleanup(struct dtx_handle *dth)
 			dtx_evict_lid(cont, dae);
 		}
 	}
-
-	vos_tx_end(dth, vos_cont2umm(cont), -DER_CANCELED);
```

Comment:
The vos_dtx_cleanup() may be called by object module directly under the case of some failure happened between two VOS API calls that belong to the same DTX. At that time, the PMDK transaction needs to be aborted. One use case is in the patch https://github.com/daos-stack/daos/pull/2998

Reply:
I reworked this a bit